### PR TITLE
Create full student application form

### PR DESCRIPTION
**Pull request by muffy**
Form has validation both client and server side.
Fixes the type of the question_key column and cleans up some unused code.

Screen Shot 2020-07-05 at 7 49 22 PM

### CODE REVIEW

File: app/controllers/students_controller.rb [hunk 1]
```diff
@@ -56,15 +77,15 @@ def waiting_list
...
+      flash.notice = "You must be an admin to view the waiting list."
     end
   end
 
   private
 
   # Only allow a list of trusted parameters through.
   def application_params
-    params.permit(:manager_email)
+    params.permit([:manager_email] + APPLICATION_QUESTIONS)
```
Comment (shaolinhoward) [inline app/controllers/students_controller.rb:88]: nice! cc: @Robert-Border

File: app/controllers/students_controller.rb [hunk 2]
```diff
@@ -74,4 +95,14 @@ def set_open_course_session
   def set_student
     @student = CourseSessionParticipant.student.find_or_initialize_by(user: current_user, course_session: @course_session)
   end
+
+  def set_answers
+    @answers = APPLICATION_QUESTIONS.map do |question|
+      answer = ApplicationAnswer.find_or_initialize_by(question_key: question, course_session_participant_id: @student.id)
+      if application_params[question].present?
+        answer.answer = application_params[question]
```
Comment (shaolinhoward) [inline app/controllers/students_controller.rb:103]: `answer.answer` 😀

File: app/models/application_answer.rb [hunk 3]
```diff
@@ -1,5 +1,3 @@
 class ApplicationAnswer < ApplicationRecord
-  belongs_to :application_question
-  belongs_to :session_participant
-  belongs_to :student
+  belongs_to :course_session_participant
```
Comment (shaolinhoward) [inline app/models/application_answer.rb:2]: hah I wonder what our model diagram looks like now

File: db/migrate/20200705185559_drop_application_questions.rb [hunk 4]
```diff
@@ -0,0 +1,7 @@
+class DropApplicationQuestions < ActiveRecord::Migration[6.0]
+  def change
```
Comment (shaolinhoward) [inline db/migrate/20200705185559_drop_application_questions.rb:2]: thanks for cleaning up!

File: app/controllers/students_controller.rb (hunk 2 above)
Comment (muffy) [reply app/controllers/students_controller.rb:103]: I KNOW!

**Pull request merged**